### PR TITLE
Update dependencies: March 2023 (TS 5.0)

### PR DESCRIPTION
**Pull request by timotheeguerin**
Generic dependency update. Notable:
-Typecript 5.0

Decide if we want this:
- change in ts api causing change in import order https://github.com/simonhaenisch/prettier-plugin-organize-imports/issues/94

Resolved 
- a bug in typedoc add me comment 2 function argument types

### CODE REVIEW

File: packages/compiler/test/checker/relation.test.ts [hunk 1]
```diff
@@ -2,9 +2,9 @@ import { deepStrictEqual, ok, strictEqual } from "assert";
 import { Model } from "../../core/index.js";
 import {
   BasicTestRunner,
+  DiagnosticMatch,
```
Comment (timotheeguerin) [inline packages/compiler/test/checker/relation.test.ts:5]: All those import change have made capitalize case ahead of  lower when sorting.
I think this is maybe better this way but does add a decent amount of changes. We unfortunatelly can't upgrade if we don't take this change right now as there is no option exposing this configuration. Linked an issue in description for the prettier organize import plugin

Comment (nguerrera) [reply packages/compiler/test/checker/relation.test.ts:5]: It's unfortunate to have all the churn, but it's prettier in my subjective opinion 😆

File: packages/compiler/test/checker/templates.test.ts [hunk 2]
```diff
@@ -124,7 +124,7 @@ describe("compiler: templates", () => {
     const { A } = (await testHost.compile("main.tsp")) as { A: Model };
     const a = A.properties.get("a")!;
     strictEqual(a.type.kind, "Scalar");
-    strictEqual((a.type as Model).name, "string");
+    strictEqual(a.type.name, "string");
```
Comment (nguerrera) [inline packages/compiler/test/checker/templates.test.ts:127]: Ooh, this means the places where did `strictEqual(a.kind, "b" as const)` can be more natural `strictEuqal(a.kind, "b")` now too, right?

File: packages/compiler/core/types.ts [hunk 3]
```diff
@@ -1916,7 +1915,9 @@ export interface EmitContext<TOptions extends object = Record<string, never>> {
    *
    * @param TypeEmitterClass The TypeEmitter to construct your emitted output
    */
-  getAssetEmitter<T>(TypeEmitterClass: typeof TypeEmitter<T, TOptions>): AssetEmitter<T, TOptions>;
+  getAssetEmitter<T>(
+    TypeEmitterClass: any /** typeof TypeEmitter<T, TOptions> https://github.com/TypeStrong/typedoc/issues/2220*/
```
Comment (nguerrera) [inline packages/compiler/core/types.ts:1919]: Curious if /** @ignore */ works here too?

File: packages/compiler/test/checker/templates.test.ts (hunk 2 above)
Comment (timotheeguerin) [reply packages/compiler/test/checker/templates.test.ts:127]: I think so

Comment (timotheeguerin) [reply packages/compiler/test/checker/templates.test.ts:127]: there is also a lot of place with the lib definition for example where we can use the new `<const T>` template param so we don't need to use `as const`

File: packages/compiler/core/types.ts (hunk 3 above)
Comment (timotheeguerin) [reply packages/compiler/core/types.ts]: ok everything was updated and it is back to working so removed the workaround.

File: packages/compiler/core/node-host.ts [hunk 4]
```diff
@@ -36,7 +36,7 @@ export const NodeHost: CompilerHost = {
   },
   getSourceFileKind: getSourceFileKindFromExt,
   logSink: createConsoleSink(),
-  mkdirp: (path: string) => mkdirp(path),
+  mkdirp: (path: string) => mkdirp(path) as any,
```
Comment (nguerrera) [inline packages/compiler/core/node-host.ts:39]: What happened here?

File: packages/compiler/emitter-framework/asset-emitter.ts [hunk 5]
```diff
@@ -387,7 +387,7 @@ export function createAssetEmitter<T, TOptions extends object>(
     emitModelProperties(model) {
       const res = typeEmitter.modelProperties(model);
       if (res instanceof EmitterResult) {
-        return res;
+        return res as any;
```
Comment (nguerrera) [inline packages/compiler/emitter-framework/asset-emitter.ts:390]: Also curios about here.

**Review by nguerrera — APPROVED** [review]

File: packages/compiler/core/node-host.ts (hunk 4 above)
Comment (timotheeguerin) [reply packages/compiler/core/node-host.ts:39]: the typing for that library is returning `Promise<undefined | void>` (void) causing issue. this triggered me also to create that remove mkdirp library issue as it is useless now

**Pull request merged**